zyte_common_items/processors.py

```diff
@@ -31,6 +31,26 @@
 )
 
 
+def _to_int(value: Any) -> Any:
+    if isinstance(value, Integral):
+        return int(value)
+    elif isinstance(value, str):
+        if "," in value:
+            value = value.replace(",", "")
+        return int(value)
+    return value
+
+
+def _to_float(value: Any) -> Any:
```

Comment:
I wonder if we should call it something different, like `to_rating_float`, to make it somewhat clear that it should not be used for values that could be greater than 999.

Reply:
What about moving these as local functions within `rating_processor`? Then it would be clear their usage is restricted to this function.
```python
def rating_processor(value: Any, page: Any) -> Any:
    def _to_int(value: Any) -> Any:
        ...

    def _to_float(value: Any) -> Any:
        ...

    ...
```
That was my original idea but I've thought that perhaps we could re-use these in some other place so I left them at the top level.

Reply:
I think renaming is better as we can still reuse the renamed ones if we want.

Reply:
@wRAR what naming would you propose...? Because `to_rating_float` sounds like something that should be used only for, well, rating.

Reply:
I agree that these functions could be useful elsewhere. I suggested `to_rating_float`, but something more generic, like `to_small_float`, would also work.

Reply:
Wouldn't guess from name alone what a function `to_small_float` does.

Reply:
I’m open to other names.

But we don’t really need for the name to make it obvious what the function can do, I think the most important think is for the name to make people think it twice before they use the function for a value that could surpass 999. The details can be explained in a docstring.

Reply:
FWIW, we also have https://github.com/zytedata/zyte-parsers/blob/8b161cf69df8caeb145e5244601ff93e41b287e6/zyte_parsers/aggregate_rating.py#L101 which does a similar thing for floats. I like how ints are handled in this PR though.

Reply:
The function in zyte-parsers also swallows the exceptions, and here it's not; it seems not silencing them is better for the case processor is processing the value returned by human-written code.